src/actions/guides/authentication/api_authentication.cr

```diff
@@ -0,0 +1,162 @@
+      -H "Authorization: Bearer fake-token"
+    ```
+
+    Using an `auth_token` query param:
+
+    ```bash
+    curl localhost:5000/api/some_endpoint?auth_token=fake-token
+    ```
+
+    Using a body param:
+
+    ```bash
+    curl localhost:5000/api/some_endpoint \
+      -d "auth_token=fake-token"
+    ```
+
+    > You could also include the token in JSON params: `{"auth_token":"fake-token"}`
+
+    #{permalink(ANCHOR_OPTIONAL_TOKEN)}
+    ## Allowing requests without a token
+
+    For actions that do not require an authentication token, include the
+    `Api::Auth::SkipRequireAuthToken` mixin defined in
+    `src/actions/mixins/api/auth/skip_require_auth_token.cr`.
+
+    For example:
+
+    ```crystal
+    class Api::PublicComments::Index < ApiAction
+      include Api::Auth::SkipRequireAuthToken
+    end
+    ```
+
+    ### Allowing unauthenticated requests by default
+
+    If most of your actions do not require an auth token you can remove
+    `Api::Auth::RequireAuthToken` from `ApiAction`.
+
+    Then in your actions that *do* require authentication, add
+    `Api::Auth::RequireAuthToken`:
+
+    ```crystal
+    class Api::SecretProjects::Index < ApiAction
+      include Api::Auth::RequireAuthToken
+    end
+    ```
+
+    ## Sign up
+
+    The `Api::SignUps::Create` action in `src/actions/api/sign_ups/create.cr`
+    handles user sign ups. It requires an email, password, and password_confirmation
+    and returns a token for authenticating the user.
+
+    ### Curl example
+
+    ```bash
+    curl localhost:5000/api/sign_ups \
+      -X POST \
+      -d "user:email=person@example.com" \
+      -d "user:password=password" \
+      -d "user:password_confirmation=password"
+    ```
+
+    Which returns a token like:
+
+    ```json
+    {"token":"some-long-token"}
+    ```
+
+    ### Customizing sign up
+
+    To add additional fields, change validations, or customize sign up,
+    check out the `SignUpUser` operation in `src/operations/sign_up_user.cr`.
+
+    ## Sign in (generate an auth token)
+
+    The `Api::SignIns::Create` action in `src/actions/api/sign_ins/create.cr`
+    handles creating a new token. It requires an email and password
+    and returns a token for authenticating the user.
+
+    ### Curl example
+
+    ```bash
+    curl localhost:5000/api/sign_ins \
+      -X POST \
+      -d "user:email=person@example.com" \
+      -d "user:password=password"
+    ```
+
+    Which returns a token like:
+
+    ```json
+    {"token":"some-long-token"}
+    ```
+
+    ### Customizing sign in
+
+    To add additional attributes, change validations, or customize sign in,
+    check out the `SignInUser` operation in `src/operations/sign_in_user.cr`.
+
+    ## Show current user
+
+    The `Api::Me::Show` in `src/actions/api/me/show.cr` returns the user for
+    the given token. Use the token returned by sign in or sign up in either
+    the `Authorization` header or in an `auth_token` param.
```

Comment:
@edwardloveall does this make sense now?

Reply:
Yes! Nicely worded.